addons/sales_team/models/crm_team.py

```diff
@@ -26,16 +34,213 @@ def _get_default_team_id(self, user_id=None):
+        return self.env.cr.dictfetchall()
+
+    def _get_graph(self):
+        def get_week_name(start_date, locale):
+            """ Generates a week name (string) from a datetime according to the locale:
+                E.g.: locale    start_date (datetime)      return string
+                      "en_US"      November 16th           "16-22 Nov"
+                      "en_US"      December 28th           "28 Dec-3 Jan"
+            """
+            if (start_date + relativedelta(days=6)).month == start_date.month:
+                short_name_from = format_date(start_date, 'd', locale=locale)
+            else:
+                short_name_from = format_date(start_date, 'd MMM', locale=locale)
+            short_name_to = format_date(start_date + relativedelta(days=6), 'd MMM', locale=locale)
+            return short_name_from + '-' + short_name_to
+
+        self.ensure_one()
+        values = []
+        today = date.today()
+        start_date, end_date = self._graph_get_dates(today)
+        graph_data = self._graph_data(start_date, end_date)
+
+        # line graphs and bar graphs require different labels
+        if self.kanban_dashboard_graph_type == 'line': # add a label ? (count / amount_untaxed) (type for bars ? => odoo color primary)
+            x_field = 'x'
+            y_field = 'y'
+        else:
+            x_field = 'label'
+            y_field = 'value'
+
+        # generate all required x_fields and update the y_values where we have data for them
+        locale = self._context.get('lang', 'en_US')
+        if self.kanban_dashboard_graph_group == 'day':
+            for day in range(0, (end_date - start_date).days + 1):
+                # name = format_date(start_date, 'd LLLL Y', locale=locale)
+                short_name = format_date(start_date + relativedelta(days=day), 'd MMM', locale=locale)
+                values.append({x_field: short_name, y_field: 0}) # , 'name': name
+            for data_item in graph_data:
+                index = (datetime.strptime(data_item.get('x_value'), DF).date() - start_date).days
+                values[index][y_field] = data_item.get('y_value')
+
+        elif self.kanban_dashboard_graph_group == 'week':
+            for week in range(0, (end_date.isocalendar()[1] - start_date.isocalendar()[1]) % 52 + 1):
+                short_name = get_week_name(start_date + relativedelta(days=7 * week), locale)
+                values.append({x_field: short_name, y_field: 0}) # 'name': name
+
+            for data_item in graph_data:
+                # FIXME: once every 28 years, this will mess up :/ and we'll have 2 values for a same x_value (2nd one will override), and none for the last
+                # why ? leap years, 1 every 7 will have 53 weeks -> on the next year 53-53%52=0, and 1-53%52=0
+                # max period is a year so no problem there
+                index = int((data_item.get('x_value') - start_date.isocalendar()[1]) % 52)
```

Comment:
Haven't seen such a method sadly. But this needs refactoring indeed.

note to self: http://stackoverflow.com/questions/21140371/get-week-number-using-date-in-python